js/build.js

```diff
@@ -1555,6 +1481,113 @@ Fliplet.Widget.instance('chat', function (data) {
     });
   }
 
+  function gatherMessageInformation(text, files) {
+    var messageData = {
+      guid: Fliplet.guid(),
+      body: text,
+      file: files ? files.file : [],
+      fileType: files ? files.fileType : undefined,
+      imageWidth: files ? files.imageWidth : undefined,
+      imageHeight: files ? files.imageHeight : undefined,
+      sentTime: new Date(),
+      sent: false,
+      conversationId: currentConversation.id,
+      widgetInstanceId: data.id,
+      appId: Fliplet.Env.get('appId'),
+      pageId: Fliplet.Env.get('pageId')
+    };
+
+    queue.push(messageData);
+    // Saves new message in QUEUE
+    return Fliplet.Storage.set(QUEUE_MESSAGE_KEY, queue.getAllQueue()).then(function () {
+      return Promise.resolve(messageData);
+    });
+  }
+
+  function changeUIOnMessageSent($parentHolder, $holder) {
+    // UI CHANGES
+    $messageArea.val('');
+    autosize.update($messageArea);
+    $parentHolder.removeClass('ready');
+    $holder.removeClass('sending');
+    resetImages();
+
+    // RETURN
+    return Promise.resolve();
+  }
+
+  function processMessage() {
+    var unsentMessages = queue.getUnsent();
+    var sendReqPromises = [];
+
+    if (Fliplet.Navigator.isOnline()) {
+      unsentMessages.forEach(function (message) {
+        sendReqPromises.push(chat.message(currentConversation.id, message));
+      });
+
+      queue.sended(unsentMessages);
```

Comment:
`sended` doesn't seem to be defined in the queue anymore. Did you actually test this code?

Reply:
@squallstar Fixed